src/disposable/types.ts

```diff
@@ -1,3 +1,7 @@
 export interface IDisposable {
   dispose(): void;
 }
+
+export interface IAsyncDisposable {
+  dispose(): void | Promise<void>;
```

Comment:
```suggestion
  dispose(): Promise<void>;
```
Shall we stick to `Promise<void>` If void is to be returned, we can use `IDisposable`.

Reply:
I want `AsyncDisposableCollection.push(IDisposable)` to work.

Reply:
Really I just want `IDisposable` to work anywhere that `IAsyncDisposable` is used. Sadly TypeScript doesn't have an `Awaitable` type that I'm aware of. Because it's true that if someone use `.then` on a Disposable.dispose() that would break :-(